docs/client-sdk/javascript/tutorials/build-an-xmtp-hello-world-app.md

```diff
@@ -0,0 +1,574 @@
+
+   type MessageListProps = {
+     msg: DecodedMessage[];
+   };
+
+   function App() {
+     const { isConnected } = useAccount()
+     const [messages, setMessages] = useState<DecodedMessage[]>([]); 
+     const [client, setClient] = useState<any>();
+     const [xmtpClientAddress, setXmtpClientAddress] = useState<any>();
+
+     const initXmtp = async function() {
+       const signer = await fetchSigner()
+       const xmtp = await Client.create(signer, { env: "production" })
+     
+       const conversation = await xmtp.conversations.newConversation(
+         PEER_ADDRESS
+       )
+       const messages = await conversation.messages({
+         direction: SortDirection.SORT_DIRECTION_DESCENDING
+       })
+       
+       setClient(conversation)
+       setMessages(messages)
+       setXmtpClientAddress(xmtp.address)
+     }
+
+     useEffect(() => {
+       if (xmtpClientAddress) {
+         const streamMessages = async () => {
+           const newStream = await client.streamMessages();
+           for await (const msg of newStream) {
+             setMessages(prevMessages => {
+               const messages = [...prevMessages];
+               messages.unshift(msg);
+               return messages;
+             });
+           }
+         };
+         streamMessages();
+       }
+     }, [client, xmtpClientAddress]);
+
+     const onSendMessage = async () => {
+       const message = "gm XMTP bot!";
+       await client.send(message)
+     }
+     
+     const MessageList: React.FC<MessageListProps> = ({ msg }) => {    
+       return (
+           <ul className="list-none overflow-y-auto h-96 flex flex-col-reverse border-2 border-sky-500 p-3 mt-8">
+             {msg.map((message, index) => (
+               <>
+                 <li key={index}>
+                   {message.senderAddress === PEER_ADDRESS ? 
+                     <span className="font-semibold underline text-red-600">BOT:</span>
+                   :
+                   <span className="font-semibold underline text-green-600">Me: </span>
+                   }
+                   {message.content}
+                 </li>
+               </>
+             ))}
+             
+           </ul>
+       )
+     };
+
+     return (
+       <div className="container flex flex-col mt-8">
+         <Web3Button/>
+         <Web3Modal
+           projectId="<YOUR_PROJECT_ID>"
+           ethereumClient={ethereumClient}
+         />
+
+         {isConnected && xmtpClientAddress &&
+           <>
+             <MessageList msg={messages} />
+             <button className="rounded-full bg-green-500 mt-4 p-2" onClick={onSendMessage}>Send GM</button>
+           </>
+         }
+
+         {isConnected && !xmtpClientAddress && 
+           <button onClick={initXmtp}>Connect to XMTP</button>}
+       </div>
+     );
+   }
+
+   export default App;
+
+
+   
+   ```
+
+if all goes well, you should have an app that looks like this:
+![gif bot tutorial](img/gm-bot.gif)
+
```

Comment:
Because this last step overwrites all of the code at once - we may need to remind people to replace `<YOUR_PROJECT_ID>` with their Product ID again.

Or maybe the first time you talk about the Project ID - you can give people a heads-up that as they work through the tutorial and start copying and pasting code - they need to remember to update the <YOUR_PROJECT_ID> value.

Reply:
good catch! Will update with an :::important. 

^ are those good mardkwon callouts to remind readers? or should we use a different type of callout?

Reply:
Yes - love callouts for something like this! =)

Here is a list of the different types: https://github.com/xmtp/xmtp-dot-org/pull/62/files#diff-38e3f93aecd4dcc78b0aab8c9b1f4220acf634f17faf508ea04cb75a70b1a551R9

You can see how they display here: https://junk-range-possible-git-markdown-stylesheet-xmtp-labs.vercel.app/markdown-test/markdown-stylesheet#calloutsadmonitions-h2

Reply:
awesome, Added a :::Info callout